Sources/Observation.swift

```diff
@@ -95,6 +99,30 @@ extension Defaults {
 		public func invalidate() {
 			object?.removeObserver(self, forKeyPath: key, context: nil)
 			object = nil
+			lifetimeTie = nil
+		}
+
+		private struct LifetimeTie {
+			weak var object: AnyObject?
+		}
+
+		private var lifetimeTie: LifetimeTie?
+		private static var lifetimeTieAssociationKey = AssociatedObject<UserDefaultsKeyObservation>()
+
+		public func tieToLifetimeOf(_ weaklyHeldObject: AnyObject) -> DefaultsObservation {
+			UserDefaultsKeyObservation.lifetimeTieAssociationKey[weaklyHeldObject] = self
+			lifetimeTie = LifetimeTie(object: weaklyHeldObject)
+			return self
+		}
+
+		public func removeLifetimeTie() {
```

Comment:
This also needs a test.

Reply:
I tried but couldn't get ARC to reliably kill the object during the test's lifetime. I tried an autoreleasepool block, waiting, etc. but the test kept working locally and failing in the Travis build. I can keep trying though.

Reply:
Never mind, I was overthinking it. :D

Just wait a while and test a weak reference for `nil`.